sql/core/src/main/scala/org/apache/spark/sql/execution/ExistingRDD.scala

```diff
@@ -175,10 +175,11 @@ private[sql] case class DataSourceScan(
   }
 
   private def canProcessBatches(): Boolean = {
+    lazy val conf = SQLContext.getActive().get.conf
     relation match {
       case r: HadoopFsRelation if r.fileFormat.isInstanceOf[ParquetSource] &&
-        SQLContext.getActive().get.conf.getConf(SQLConf.PARQUET_VECTORIZED_READER_ENABLED) &&
-        SQLContext.getActive().get.conf.getConf(SQLConf.WHOLESTAGE_CODEGEN_ENABLED) =>
+        conf.parquetVectorizedReaderEnabled &&
+        conf.wholeStageEnabled && schema.length <= conf.wholeStageMaxNumFields =>
```

Comment:
is it because we have to synchronize this with whole stage codegen because we don't have great planner support for this?